src/coreclr/jit/helperexpansion.cpp

```diff
@@ -478,36 +478,54 @@ bool Compiler::fgExpandThreadLocalAccessForCall(BasicBlock** pBlock, Statement*
         return false;
     }
 
+#if defined(FEATURE_READYTORUN)
+    assert(!"Unsupported scenario of optimizing TLS access for ReadyToRun");
+#endif
```

Comment:
Should you be using `Compiler::IsReadyToRun()` instead?

Reply:
probably